cpp/CMakeLists.txt

```diff
@@ -256,8 +277,18 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
 ###################################################################################################
 # - find zlib -------------------------------------------------------------------------------------
 
-find_package(ZLIB REQUIRED)
+find_package(ZLIB REQUIRED QUIET)
+
+if (NOT ZLIB_FOUND)
+  CPMFindPackage(
+    NAME ZLIB
+    GITHUB_REPOSITORY madler/zlib
+    GIT_TAG 1.2.11
+    VERSION 1.2.11
+  )
+endif()
```

Comment:
Paraphrasing my comments from L389

```
  # This method detects if it's local, but if it is not available uses CPM to install the package.
  # By default CPM does not add the standard variables that come with find package such as `GTEST_LIBRARY_DIR`.
  # So we detect then use `CPMFind/AddPackage` and set the library dir ourselves. 
  # Possibly this is worth patching to support but I don't fully grok the design decision yet.
```
We should not need to, but at the phase I'm in I've been wading through import and linking strategies. This is here so I can test out this pattern with a know good library that is importing and building properly and I can verify the linking. But yes, you are correct. I'm 90% of the way to adding a wrapper for CPM to set standard vars for imported kit.

While typing this I realize I didn't link to the docs on that and possibly the variables are set with Find but not Add. Will check and solve. 

In summary; yes, you are correct this is not what should be done.

Reply:
Got it. Yea this is because doing a `find_package(ZLIB)` hits a CMake specific `FindZLib` module: https://cmake.org/cmake/help/v3.18/module/FindZLIB.html

Reply:
Not quite related, but this code snippet won't work as intended because of the `REQUIRED` in the `find_package()`: If zlib isn't found, the config will error out right there.